cpp/tests/link_prediction/mg_weighted_similarity_test.cpp

```diff
@@ -0,0 +1,470 @@
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle_->get_comms().barrier();
+      hr_timer.start("MG Construct graph");
+    }
+
+    auto [mg_graph, mg_edge_weights, d_mg_renumber_map_labels] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, true>(
+        *handle_, input_usecase, true, true, false, true);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle_->get_comms().barrier();
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    // 2. run similarity
+
+    auto mg_graph_view = mg_graph.view();
+    auto mg_edge_weight_view =
+      mg_edge_weights ? std::make_optional((*mg_edge_weights).view()) : std::nullopt;
+
+    rmm::device_uvector<vertex_t> d_start_vertices(
+      std::min(
+        static_cast<size_t>(mg_graph_view.local_vertex_partition_range_size()),
+        similarity_usecase.max_seeds / comm_size +
+          (static_cast<size_t>(comm_rank) < similarity_usecase.max_seeds % comm_size ? 1 : 0)),
+      handle_->get_stream());
+
+    /*
+        for (int k = 0; k < comm_size; k++) {
+          auto& comm = handle_->get_comms();
+
+          comm.barrier();
+          if (comm_rank == k) {
+            std::cout << "Rank :" << comm_rank << std::endl;
+            RAFT_CUDA_TRY(cudaDeviceSynchronize());
+
+            raft::print_device_vector(
+              "d_start_vertices", d_start_vertices.data(), d_start_vertices.size(), std::cout);
+
+            std::cout << "------------------" << std::endl;
+          }
+          comm.barrier();
+        }
+
+        */
+
+    cugraph::test::populate_vertex_ids(
+      *handle_, d_start_vertices, mg_graph_view.local_vertex_partition_range_first());
+
+    /*
+        for (int k = 0; k < comm_size; k++) {
+          auto& comm = handle_->get_comms();
+
+          comm.barrier();
+          if (comm_rank == k) {
+            std::cout << "Rank :" << comm_rank << std::endl;
+            RAFT_CUDA_TRY(cudaDeviceSynchronize());
+
+            raft::print_device_vector(
+              "d_start_vertices", d_start_vertices.data(), d_start_vertices.size(), std::cout);
+
+            std::cout << "------------------" << std::endl;
+          }
+          comm.barrier();
+        }
+        */
+
+    auto [d_offsets, two_hop_nbrs] = cugraph::k_hop_nbrs(
+      *handle_,
+      mg_graph_view,
+      raft::device_span<vertex_t const>(d_start_vertices.data(), d_start_vertices.size()),
+      2);
+
+    auto h_start_vertices = cugraph::test::to_host(*handle_, d_start_vertices);
+    auto h_offsets        = cugraph::test::to_host(*handle_, d_offsets);
+
+    std::vector<vertex_t> h_v1(h_offsets.back());
+    for (size_t i = 0; i < h_start_vertices.size(); ++i) {
+      std::fill(h_v1.begin() + h_offsets[i], h_v1.begin() + h_offsets[i + 1], h_start_vertices[i]);
+    }
```

Comment:
These are necessary only when check_correctness is true? Right? Better define these variables inside the `if (use_case.check_correctness) {}` block.

Reply:
We need it to construct  input `vertex_pairs` even if `use_case.check_correctness` is `false`